### ISSUE TITLE
[Balance] EMP and shield.

### ISSUE THREAD

**Issue by Tree3324417** [PavelZinchenko/event-horizon-frontier-builds#121]
This suggestion covers shield balance in both Event Horizon and Event Horizon - Frontier.
- Currently, the primary factor holding shield back from being useful is the extreme vulnerability toward EMP weapons. 
While I agree that EMP weapons should remain strong against shield, the current EMP weapons (EMP torpedoes, EMP nuke) often just delete high amount of shield HPs. Ships like velez or SBm2 which utilizes extremely high shield modifier exist as a result, which in my honest opinion is not the best thing - shield should not be restricted to just one or two ships.

- Solution: Lower the power multiplier of EMP weapons against shields. EMP weapons should do around 3-5 times base damage toward shield, not the current ~10-20x. This also means the EMP nuke should be adjusted to have a more reasonable shielddrain modifier. After this, can adjust velez to have a more reasonable shield modifier (no need for Mk2 Starbase, since being both tanky and still uses weapons is it's strength, and it should not lose that). 

- Also, I request a buff to shield generators. Currently they are largely useless due to horrible space efficiency which is far more effective when spent toward weapon boosters or reactors, which also makes the shield no cooldown perk pretty much useless.

- Additional Notes: I would be glad if you, Pavel, could be active once again in your discord account. The main Event Horizon community is still present there, and it would be nice if you can be in the discord server for easier contact and talks with the player base.

**Comment by Inedilarion** [comment]
I'll add a little. The only thing in which shields are better than armor (I do not take into account generators, as mentioned above, they take up too many valuable green cells, giving a meager increase in survivability, relevant only for Starbase MK2), is weight. But at the cost of energy consumption, moreover, than stabilizers spend on the same armor area.
In what aspects are shields worse than armor?
1) EMP, as stated above.
2) Shields are not affected by bonuses to armor, resists and regeneration (this would not be a problem if there were about the same number of ships with bonuses to shields as there were bonuses to armor, not just one Velez).
3) There are no modifications to the shields at the shipyard except for accelerated regeneration, which gives already rather poor shields a powerful debuff. While for armor there are resists and regeneration.
4) To repair armor, there are rays, as well as repair bots that are much more compact than generators

As a result, shields themselves are in almost every way worse than armor, and the use of generators forces us to reduce some other characteristics of the ship too much. I can also suggest some enhancement options:
1. Add a small built-in regeneration to the shield batteries (probably about 2-5% of capacity. This is based on the minimum alternative for armor, regeneration, and the maximum, repair bot). This will give shields a second bonus besides mass compared to armor. However, generators will become even less necessary.
2. Make generators work like repair bots. So that they restore the shield as a percentage and install 1 per ship. The disadvantage of this approach will be the impossibility of making a low-capacity shield with high recovery, however, since this approach is not used by anyone anywhere, the loss is insignificant.
3. Simply increase shield regeneration in generators. In this case it is already very difficult to say how much. I recently assessed the usefulness of a shield modification for Veles in the arena, and found that to compensate for the loss of shield capacity by 2 times, its regeneration is not enough by 3-5 times. However, this is in a situation where weapon enhancements are almost completely replaced; for an adequate build, the generator enhancements should be even greater. However, with the generator strengthened 10 times, Velez will be able to get approximately 3000 shield regeneration, which will already be very difficult to destroy (2 HLC on it have only 600 DPS. 2250 with the scientist’s boosts, which even so will require significant weapon boosts to overpower the regeneration)

**Comment by TaipionsSalemClient** [comment]
I disagree with the EMP nerfs in general, it could be less but it does not need to be nerfed like suggested,
shields can use a buff, and most of what @Inedilarion said is correct, though I disagree especially on the "repair bot for shields" idea.

Instead, also to further distinguish shields from armour, I propose:

- give shields in general a zero cooldown for regeneration
- new module "shield booster" 2x2 size low cost, increases all shield regen to 10x for 10s with 60s base cooldown (low cost and small size as it only amplifies what is already there, so you need a lot of shield generators to make it useful)
- **OR** make repair bots work on shields, too
- repair rays work on shields **after** hp of the target ship/SB reached 100%
=> it would be useless to have 2 different sets of "healing equipment" for armour and shields, you'd have to fully go one direction and be handicapped if you'd mix  armour and shields ships, making things too complicated in the process, therefore simplify things and make it just one mechanic to repair both
- and add some useful shield perks, nothing with resists to prevent both mixing up things with armour and making shields distinct from armour, rather have perks that:
(some ideas)
increase shield hp by a fixed %,
increase shield regeneration,
reduce energy cost and weight of shield modules,
resistence to emp, 
"backup shields" - shields regain 50% when reaching zero, 2 minutes cooldown,
"high shields" - shield regen is 2x when over 50%, but 1/2 when under 50%,
"resilient shields" - shield regen is 2x when under 25%, mutually exclusive with "high shields"

An important point here with shield perks is, to give at least 4-5 "good" ones, better 5-6, so that there are no 3 best perks that everyone takes for shields and that's it.

**Comment by Inedilarion** [comment]
> * give shields in general a zero cooldown for regeneration
I thought about adding this to my list of suggestions, but I looked in the game at how many generators can actually be installed on different ships, and it turned out that even Velez would not be able to make generators relevant. So this change in itself is not enough (it also removes the corresponding modification at the shipyard) and will only affect Starbase MK2

**Comment by Asmageddon** [comment]
Instead of making repair ray work on shields and removing shield regen cooldown, what about:

 - Shields only absorb 80% of damage, so you still lose HP, which can be repaired, but are more efficient overall in return.
 - Constant regen at 25% speed, 100% after the cooldown.
 - Do reduce EMP shield damage somewhat.

**Comment by TaipionsSalemClient** [comment]
maybe take a look at the last version of the game... there already is a "shield repair ray" the energizer...

the idea above with that 80% really sounds ugly, no one wants that,

... with EMP staying at a reasonably high shield dmg (could still be a bit lower than what it is now) and even considering the (mostly useless) bonus against corrosive (mostly useless because of how ridiculous the AR is),
it would still not hurt to put shield to "0cd" (read: zero cooldown) by default without any tradeoff, and removing the shield perk, or reworking it in a way,
**_shields are still considerably weaker than armour due to how resists work, and emp ofc_**

possible options to rework the 0cd shield perk (and/or make new shield perks) in case this is considered:
- perk grants immunity to emp, but reduces shield hp and regen by 20%
- perk applies "adaptive resists" to shields in that: if the current hit is the same dmg type as the last hit, dmg is halved, if not, dmg is doubled
- perk reverses not-type immunity in that: no-type dmg is doubled on shields, other dmg is halved
with shield perks like these, you would have a lot of variety without it being too op, making shields a kind of a wild card and interesting again

**Comment by Asmageddon** [comment]
I don't think constant regen is the right way to balance shields - if you buff them enough, the rapid regeneration is already a great boon, and could make ships unkillable without powerful alpha strikes. I could get behind 25% constant regen though, if just because 4s is a really long time.

Instead, if it's resistances that matter, what about one of:
 - Shield capacitors come (only) in 3 variants with built-in resistances, so you can mix and match between 25% resistances across the board, 33% to two types, 50% to one, anything in-between, or more if you go for resistance mods over defense mods.
 - Buff total shield capacity and introduce green shield modulator components, which shift the shield towards resisting 1/2 damage types better at the cost of the other(s)
 - something something shield layers?

Different elemental shields should also have different benefits, e.g. probability of bouncing projectiles back, partial reflection of laser weapons, emp resistance.

**Comment by TaipionsSalemClient** [comment]
> I don't think constant regen is the right way to balance shields - if you buff them enough, the rapid regeneration is already a great boon, and could make ships unkillable without powerful alpha strikes. 

ever heard of emp / emp nuke?
